hw4/ParseTree/PTree.cs

```diff
@@ -0,0 +1,178 @@
+﻿namespace ParseTree;
+
+using System;
+
+/// <summary>
+/// Класс дерева
+/// </summary>
+public class PTree
+{
+    private INode ? root;
+
+    /// <summary>
+    /// Конструктор дерева
+    /// </summary>
+    public PTree()
+    {
+        this.root = null;
+    }
+
+    /// <summary>
+    /// Преобразование строки в число
+    /// </summary>
+    /// <param name="givenString">Данная строка</param>
+    /// <param name="index">Индекс текущего элемента</param>
+    /// <returns>Число, полученное из строки</returns>
+    public int numberConvert(string givenString, ref int index)
+    {
+        int number = 0;
+        while (index < givenString.Length && givenString[index] >= '0' && givenString[index] <= '9')
+        {
+            number = number* 10 + (int) (givenString[index] - '0');
+            ++index;
+        }
+    return number;
+    }
+
+    /// <summary>
+    /// Проверка символа на арифметическую операцию
+    /// </summary>
+    /// <param name="symbol">Данный символ</param>
+    /// <param name="next">Следующий символ</param>
+    /// <returns>Является ли текущий символ арифметической операцией</returns>
+    public bool isOperation(char symbol, char next)
+    {
+        if (symbol == '+' || symbol == '*' || symbol == '/')
+        {
+            return true;
+        }
+        if (symbol == '-')
+        {
+            return '9' - next < 0 || next - '0' < 0;
+        }
+        return false;
+    }
+
+    private INode AddNodeRecursive(string givenString, ref int index)
+    {
+        ++index;
+        while (index < givenString.Length && (givenString[index] == ' ' ||
+            givenString[index] == '(' || givenString[index] == ')'))
+        {
+            ++index;
+        }
+        INode node;
+        if (index == givenString.Length - 1)
+        {
+            if (givenString[index] - '0' >= 0 && '9' - givenString[index] > 0)
+            {
+                node = new Operand(numberConvert(givenString, ref index));
+                return node;
```

Comment:
```suggestion
                return new Operand(numberConvert(givenString, ref index));
```
Тут node вообще не нужен, его можно было так рано и не объявлять